react/features/stream-effects/blur/JitsiStreamBlurEffect.js

```diff
@@ -61,61 +65,108 @@ export default class JitsiStreamBlurEffect {
     }
 
     /**
-     * Loop function to render the background mask.
+     * Represents the run post processing.
      *
-     * @private
      * @returns {void}
      */
-    async _renderMask() {
-        if (!this._maskInProgress) {
-            this._maskInProgress = true;
-            this._bpModel.segmentPerson(this._inputVideoElement, {
-                internalResolution: 'low', // resized to 0.5 times of the original resolution before inference
-                maxDetections: 1, // max. number of person poses to detect per image
-                segmentationThreshold: 0.7, // represents probability that a pixel belongs to a person
-                flipHorizontal: false,
-                scoreThreshold: 0.2
-            }).then(data => {
-                this._segmentationData = data;
-                this._maskInProgress = false;
-            });
-        }
-        const inputCanvasCtx = this._inputVideoCanvasElement.getContext('2d');
+    runPostProcessing() {
+        this._outputCanvasCtx.globalCompositeOperation = 'copy';
 
-        inputCanvasCtx.drawImage(this._inputVideoElement, 0, 0);
-
-        const currentFrame = inputCanvasCtx.getImageData(
+        // draw segmentation mask
+        this._outputCanvasCtx.filter = 'blur(25px)';
+        this._outputCanvasCtx.drawImage(
+            this._segmentationMaskCanvas,
+            0,
+            0,
+            segmentationWidth,
+            segmentationHeight,
             0,
             0,
-            this._inputVideoCanvasElement.width,
-            this._inputVideoCanvasElement.height
+            this._inputVideoElement.width,
+            this._inputVideoElement.height
         );
 
-        if (this._segmentationData) {
-            const blurData = new ImageData(currentFrame.data.slice(), currentFrame.width, currentFrame.height);
+        this._outputCanvasCtx.globalCompositeOperation = 'source-in';
+        this._outputCanvasCtx.filter = 'none';
+        this._outputCanvasCtx.drawImage(this._inputVideoElement, 0, 0);
 
-            StackBlur.imageDataRGB(blurData, 0, 0, currentFrame.width, currentFrame.height, 12);
-
-            for (let x = 0; x < this._outputCanvasElement.width; x++) {
-                for (let y = 0; y < this._outputCanvasElement.height; y++) {
-                    const n = (y * this._outputCanvasElement.width) + x;
+        this._outputCanvasCtx.globalCompositeOperation = 'destination-over';
+        this._outputCanvasCtx.filter = 'blur(25px)'; // FIXME Does not work on Safari
```

Comment:
Can you double check how this works in Safari?

Reply:
I looked over this yesterday and unfortunately, Context.filter is not supported in Safari. Maybe we can take a look [here](https://www.npmjs.com/package/context-filter-polyfill)